File: Digital_Library_Web_Application_Fullstack/User-frontend/src/app/models/bookrental.ts
import { Book } from "./book";
import { SignUp } from "./sign-up";

export class BookRental {
    id: number = 0;
    rentalDate: string = "";
    dueDate: string = "";
    actualReturnDate: string = "";
    rentAmount: number = 0;
    hasReturned: boolean = false;
    fineAmount: number = 0;
    book:Book = new Book();
    user:SignUp = new SignUp()

    constructor(
        id:number,
        rentalDate:string,
        dueDate:string,
        actualReturnDate:string,
        rentAmount:number,
        hasReturned: boolean,
        fineAmount: number,
        book:Book,
        user: SignUp
    )
    {
        this.id = id;
        this.rentalDate = new Date(rentalDate).toLocaleDateString();
        this.dueDate = dueDate;
        this.actualReturnDate = actualReturnDate;
        this.rentAmount = rentAmount;
        this.hasReturned = hasReturned;
        this.fineAmount = fineAmount;
        this.book = book;
        this.user = user;
    }
}

File: Digital_Library_Web_Application_Fullstack/Back-end/Library.Data/Entities/BookRental.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Library.Data.Entities
{
    public class BookRental : BaseObject
    {
        public Book Book { get; set; }
        public User User { get; set; }

        [Required]
        public DateTime RentalDate { get; set; }
        public DateTime? DueDate { get; set; }
        public DateTime? ActualReturnDate { get; set; }
        public float? RentAmount { get; set; }
        public bool HasReturned { get; set; }
        public float? FineAmount { get; set; }

    }

}